smarts/core/smarts.py

```diff
@@ -234,6 +241,8 @@ def _step(self, agent_actions):
 
         observations, rewards, scores, dones = self._agent_manager.observe(self)
 
+        self.check_if_acting_on_non_existing_agent(agent_actions, dones)
```

Comment:
I would suggest trying this out using more than one agent to see if this works properly.

Reply:
True, this wouldn't work for multi-agent case
when running:
``` python examples/multi_agent.py scenarios/intersections/6lane```

```
WARNING:SMARTS:Attempted giving action to an non-existing agent, Agent 0 
WARNING:SMARTS:Attempted giving action to an non-existing agent, Agent 1 
WARNING:SMARTS:Attempted giving action to an non-existing agent, Agent 3 
│               0/10 │               0.79 │                 59 │               7.92 │              6lane │                    │ -65129523927046146 │   162.84 - Agent 2 │
pybullet build time: Dec 23 2020 01:47:29
pybullet build time: Dec 23 2020 01:47:29
pybullet build time: Dec 23 2020 01:47:29
pybullet build time: Dec 23 2020 01:47:29
WARNING:SMARTS:Attempted giving action to an non-existing agent, Agent 3 
WARNING:SMARTS:Attempted giving action to an non-existing agent, Agent 0 
WARNING:SMARTS:Attempted giving action to an non-existing agent, Agent 1
```

Reply:
@JingfeiPeng The example you gave above, actually is called "multi scenarios". "Multi agents" means we have multiple agents in the example.

Reply:
I think this `check_if_acting_on_non_existing_agent` is not working as we expected, because we are calling it after `observe`, i.e., the `dones` are calculated based on the current `agent_actions`, and seems to be against our intention.